From 63981d8948d468a428d06cd926cb0623b01c8141 Mon Sep 17 00:00:00 2001
From: Codey <codey@eliterepairs.com.au>
Date: Tue, 27 Oct 2026 16:41:20 +1000
Subject: [PATCH 08/31] PS4 LED: always send the colour command (boot ICC sets
 blue behind the driver's back); southbridge log line without the doubled
 prefix

---
 arch/x86/platform/ps4/ps4.c | 2 +-
 drivers/ps4/ps4-led.c       | 7 +++----
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/arch/x86/platform/ps4/ps4.c b/arch/x86/platform/ps4/ps4.c
index f0aed8893..54f3da311 100644
--- a/arch/x86/platform/ps4/ps4.c
+++ b/arch/x86/platform/ps4/ps4.c
@@ -125,7 +125,7 @@ int __ref ps4_is_baikal_early(void)
 
 	ps4_baikal_cached = ((id & 0xffff) == PCI_VENDOR_ID_SONY &&
 			     ((id >> 16) & 0xffff) == PCI_DEVICE_ID_SONY_BAIKAL_ACPI);
-	pr_info("ps4: southbridge id %08x, Baikal paths %s\n", id,
+	pr_info("southbridge id %08x, Baikal paths %s\n", id,
 		ps4_baikal_cached ? "on" : "off");
 	return ps4_baikal_cached;
 }
diff --git a/drivers/ps4/ps4-led.c b/drivers/ps4/ps4-led.c
index ded03d60a..bd5167e7d 100644
--- a/drivers/ps4/ps4-led.c
+++ b/drivers/ps4/ps4-led.c
@@ -132,10 +132,9 @@ static int ps4_led_set_blocking(struct led_classdev *led_cdev,
 
 	mutex_lock(&ps4_led_lock);
 
-	if (ps4_led_current_payload == data) {
-		mutex_unlock(&ps4_led_lock);
-		return 0;
-	}
+	/* Always send: the ICC boot sequence sets the LED itself (solid blue once
+	 * the kernel is up) without this driver knowing, so a cached "already
+	 * this colour" check would wrongly skip the request. */
 
 	ret = apcie_icc_cmd(PS4_LED_ICC_MAJOR, PS4_LED_ICC_MINOR,
 			    data, PS4_LED_PAYLOAD_LEN,
